### PR TITLE
src: update knative deps to 0.22

### PR DESCRIPTION
**Pull request by matejvasek**
Signed-off-by: Matej Vasek <mvasek@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,26 +5,28 @@ go 1.14
...
+	k8s.io/client-go v0.19.7
+	knative.dev/client v0.22.0
+	knative.dev/eventing v0.22.0
+	knative.dev/pkg v0.0.0-20210331065221-952fdd90dbb0
+	knative.dev/serving v0.22.0
 )
 
 replace (
+	// knative.dev/serving@v0.22.0 and knative.dev/pkg@v0.0.0-20210331065221-952fdd90dbb0 require different versions of go-openapi/spec
+	github.com/go-openapi/spec => github.com/go-openapi/spec v0.19.6
```
Comment (matejvasek) [inline go.mod:28]: @dsimansk would this work with `kn`?
For some reason `knative.dev/serving@v0.22.0` has indirect dependency on `github.com/go-openapi/spec v0.20.2`, but that breaks something for `sigs.k8s.io/kustomize`.

**Comment by matejvasek** [comment]
this improves cancellability

**Review by zroubalik — COMMENTED** [review]
I am generally ok with this change. But OpenShift Serverless 1.14 uses Knative 0.20.0, the next version will probably use 0.21.0? But we are bringing dependency on 0.22.0 in here.

Maybe it is ok, but I am not sure 🤷‍♂️

**Comment by matejvasek** [comment]
@rhuss wdyt?

File: client.go [hunk 2]
```diff
@@ -101,7 +101,7 @@ type ProgressListener interface {
 // Describer of Functions' remote deployed aspect.
 type Describer interface {
 	// Describe the running state of the service as reported by the underlyng platform.
-	Describe(name string) (description Description, err error)
+	Describe(ctx context.Context, name string) (description Description, err error)
```
Comment (lance) [inline client.go:104]: I'm not sure what to say about bumping knative dependency - probably @rhuss or @markusthoemmes have more insight on that. But I think the changes in this file are probably unrelated, aren't they? I do like adding context in general, but does it need to be in this PR?

Comment (matejvasek) [reply client.go:104]: I am updating the dep specifically to be able to use ctx.

Comment (matejvasek) [reply client.go:104]: but I could do separate PR

**Comment by matejvasek** [comment]
We really should use `0.21.0` for upcoming version.

**Pull request closed without merging**